### DISCUSSION TITLE
Edit Geo-Fence Charge Cost not saving

### DISCUSSION THREAD

**Discussion by natrlhy** [teslamate-org/teslamate#3889] [Q&A] [answered]
I've been trying, yet it never saves...

I go into my Home Geo-Fence. I enter the per kWh to 0.3521.
I click "Continue" (there is another button to click to Add costs retroactively but it doesn't matter which one I pick.
Screen returns to the Geo-Fence page but the cost never updates.

Any reason why this would occur or how I can updated it and have it save properly?

Thanks!

**Comment by Otsirc** [comment]
Same issue here. As workaround i update it manually in the DB via a query like UPDATE charging_processes SET cost=charge_energy_used*0.113 WHERE geofence_id='1' and start_date >= '2024-03-01 00:00:00' ;

**Reply by natrlhy** [reply]
Is there a direct query to modify the charge rate for the geo-fence ID itself (11 in my case)?

I did go ahead and run this as well:

`teslamate=# UPDATE charging_processes SET cost=charge_energy_used*0.3521 WHERE geofence_id='11' and start_date >= '2024-04-01 00:00:00' ;`
`UPDATE 16`
Thanks!

**Comment by natrlhy** [comment, marked as answer]
So after a little AI questioning I was able to figure this out! This is how I manually updated the charge `cost_per_unit ` in my `geofence` table with a name of `Home` 🥇 
```
SELECT * FROM geofences WHERE name LIKE 'Home';
UPDATE geofences SET cost_per_unit = 0.3521 WHERE id = 11;
```
Ta-da! Still uncertain why the web interface won't save it.

**Comment by adansoILT** [comment]
Same issue. Can someone fix this issue?

**Comment by djpalmis** [comment]
https://github.com/teslamate-org/teslamate/issues/4058 i opened a ticket some time ago but some very skilled software professional closed my issue 😄
